### ISSUE TITLE
[BUG]: inconsistancy between currentTime and playableDuration in onProgress callback

### ISSUE THREAD

**Issue by yahacom** [TheWidlarzGroup/react-native-video#3866]
### Version

6.1.2

### What platforms are you having the problem on?

iOS, Android

### Architecture

Old architecture

### What happened?

I have differencies between `currentTime` and `playableDuration` params from `onProgress`.
When video plays and stops at the end I expect that `currentTime` will be equal to `playableDuration`. But actual result is `currentTime` less than `playableDuration` for couple of milliseconds (dependent on video duration).
I also compare `currentTime` at the end of video with the video duration value returned from `onLoad` callback; the issue is the same.
It's become more critical for small videos less than 10 sec.

### Reproduction

repository link

### Reproduction

Step to reproduce this bug are: 
I provide simple code that shows what I do and under comments you can see what I receive in callbacks.
So, when video finishes playing I expect that `currentTime === playableDuration`
```typescript
  const handleVideoProgress = useCallback(
    ({ currentTime, playableDuration, seekableDuration }: OnProgressData) => {
      console.log(currentTime, playableDuration); // 596.2512817382812 596.473333333
    },
    [],
  );
  const handleLoad = useCallback(({ duration }: OnLoadData) => {
      console.log(duration); // 596.473333333
  }, []);
```

```jsx
        <Video
          source={{ uri: 'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4' }}
          ref={videoRef}
          style={styles.video}
          resizeMode={ResizeMode.CONTAIN}
          controls={false}
          onProgress={handleVideoProgress}
          onLoad={handleLoad}
        />
```

**Comment by Ibad9** [comment]
You can use Math.floor to avoid conflict.

**Comment by freeboub** [comment]
I disagree @Ibad9 Math.floor will just "corrupt data and behavior".
We can add a last onProgress call at the end of playback, It is not an issue.
But why doin't you use [onEnd](https://thewidlarzgroup.github.io/react-native-video/component/events#onend) callback to catch playback end ?

**Comment by freeboub** [comment]
I created a pull request to fullfil your requierment as I think it make sense, but you need to use onEnd anyway !
Please let me know if it works for you please !

**Comment by yahacom** [comment]
Hey @freeboub ! Thanks for quick reply.

In my case, I implemented custom controls, especially video progress bar, and using `onProgress` prop to retrieve current position. And every time video finishes, progress bar does not end because of this difference (especially on short videos). Maybe you could suggest a better way to do this?

Currently, I fixed it as you suggested with `onEnd` - just moving progress bar to end manually once `onEnd` is triggered.

UPD: I just checked your PR with fix. It seems to work fine. Thanks.
This issue can be closed I guess.

**Comment by freeboub** [comment]
@yahacom Yes, that's clear, I think you can just test the PR I opened, it should fix the issue, I just force sending onProgress before the onEnd.

**Comment by yahacom** [comment]
@freeboub I checked your PR with fix. It works fine.
Thanks.

**Comment by freeboub** [comment]
fix will be available in 6.2
